Loop/Managers/LoopAlertsManager.swift

```diff
@@ -50,13 +60,104 @@ public class LoopAlertsManager {
+    }
+
+    func scheduleLoopNotRunningNotifications() {
+        // Give a little extra time for a loop-in-progress to complete
+        let gracePeriod = TimeInterval(minutes: 0.5)
+
+        var scheduledNotifications: [StoredLoopNotRunningNotification] = []
+
+        for (minutes, isCritical) in [(20.0, false), (40.0, false), (60.0, true), (120.0, true)] {
+            let notification = UNMutableNotificationContent()
+            let failureInterval = TimeInterval(minutes: minutes)
+
+            let formatter = DateComponentsFormatter()
+            formatter.maximumUnitCount = 1
+            formatter.allowedUnits = [.hour, .minute]
+            formatter.unitsStyle = .full
+
+            if let failureIntervalString = formatter.string(from: failureInterval)?.localizedLowercase {
+                notification.body = String(format: NSLocalizedString("Loop has not completed successfully in %@", comment: "The notification alert describing a long-lasting loop failure. The substitution parameter is the time interval since the last loop"), failureIntervalString)
+            }
+
+            notification.title = NSLocalizedString("Loop Failure", comment: "The notification title for a loop failure")
+            if isCritical, FeatureFlags.criticalAlertsEnabled {
+                if #available(iOS 15.0, *) {
+                    notification.interruptionLevel = .critical
+                }
+                notification.sound = .defaultCritical
+            } else {
+                if #available(iOS 15.0, *) {
+                    notification.interruptionLevel = .timeSensitive
+                }
+                notification.sound = .default
+            }
+            notification.categoryIdentifier = LoopNotificationCategory.loopNotRunning.rawValue
+            notification.threadIdentifier = LoopNotificationCategory.loopNotRunning.rawValue
+
+            let trigger = UNTimeIntervalNotificationTrigger(
+                timeInterval: failureInterval + gracePeriod,
+                repeats: false
+            )
+
+            let request = UNNotificationRequest(
+                identifier: "\(LoopNotificationCategory.loopNotRunning.rawValue)\(failureInterval)",
+                content: notification,
+                trigger: trigger
+            )
+
+            if let nextTriggerDate = trigger.nextTriggerDate() {
+                let scheduledNotification = StoredLoopNotRunningNotification(
+                    alertAt: nextTriggerDate,
+                    title: notification.title,
+                    body: notification.body,
+                    timeInterval: failureInterval,
+                    isCritical: isCritical)
+                scheduledNotifications.append(scheduledNotification)
+            }
+            UNUserNotificationCenter.current().add(request)
+        }
+        UserDefaults.appGroup?.loopNotRunningNotifications = scheduledNotifications
+    }
+
+    func clearLoopNotRunningNotifications(alertStore: AlertStore) {
```

Comment:
Consider passing in `AlertManager` here, instead of `AlertStore`, to have the responsibility to record the alert

Reply:
Having an `AlertStore` at all?  Seems to me it is unnecessary to reveal it here, since we already have a precedent for forwarding `AlertStore` storage and queries.